### PR TITLE
Do not double check ACLs on scheduled reminders

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Do not double check ACLs on scheduled reminders

Before
----------------------------------------
The code gets a list of events from ` $events = \CRM_Event_BAO_Event::getEvents();` and passes it through the `\CRM_ACL_API::group(\CRM_Core_Permission::EDIT, NULL, 'civicrm_event', $events);` function to filter to those permitted by ACL. However, ` $events = \CRM_Event_BAO_Event::getEvents();` itself has a `checkPermissions = TRUE` in this scenario.

That parameter leads to `CRM_Event_BAO_Event::checkPermission($eventID)` which does the same ACL check internally here 

https://github.com/civicrm/civicrm-core/blob/db3bae353ad1823d36f5bba6b8553b331d820e3d/CRM/Event/BAO/Event.php#L2072

After
----------------------------------------
I'm just relying on the permission check from ` $events = \CRM_Event_BAO_Event::getEvents();`

Technical Details
----------------------------------------
This started off with me trying to answer @artfulrobot's question about how the paramters for the function that is now no longer called here are used, and then I wound up seeing @colemanw's hated `retrieveValue` and now both are gone - and I have to find other usages to answer @artfulrobot  :-)

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Event/BAO/Event.php [hunk 1]
```diff
@@ -307,14 +307,15 @@ public static function getEvents(
 
     $dao = CRM_Core_DAO::executeQuery($query);
     while ($dao->fetch()) {
+      $eventID = (int) $dao->id;
```
Comment (eileenmcnaughton) [inline CRM/Event/BAO/Event.php:310]: cast to int

File: Civi/API/Subscriber/PermissionCheck.php [hunk 2]
```diff
@@ -105,14 +106,9 @@ public function checkACLPermission($apiRequest) {
...
-        $aclEdit = \CRM_ACL_API::group(\CRM_Core_Permission::EDIT, NULL, 'civicrm_event', $events);
-        $param = ['id' => $apiRequest['params']['id']];
-        $eventId = \CRM_Core_BAO_ActionSchedule::retrieve($param, $value = []);
-        if (in_array($eventId->entity_value, $aclEdit)) {
-          return TRUE;
-        }
-        break;
+        $permittedEvents = \CRM_Event_BAO_Event::getEvents();
+        $eventId = (int) \CRM_Core_DAO::getFieldValue('CRM_Core_BAO_ActionSchedule', $apiRequest['params']['id'], 'entity_value');
+        return (in_array($eventId, $permittedEvents, TRUE));
```
Comment (eileenmcnaughton) [inline Civi/API/Subscriber/PermissionCheck.php:111]: can add `TRUE` as I ensured it is cast to an int

**Comment by seamuslee001** [comment]
@JoeMurray this looks at least partially linked to our client's db issues

**Comment by eileenmcnaughton** [comment]
@seamuslee001 it is pretty narrow & really just touches ACLs via the api on Scheduled Reminders

**Comment by JoeMurray** [comment]
I think this is the right thing to do but would not that it means that a user able to create a scheduled reminder can cause an email or sms to be sent to contacts that they themselves are not allowed to view.

**Comment by eileenmcnaughton** [comment]
Right - this bit of code says 'do not let someone create a scheduled reminder for an event they don't have permissions on' 

TBH I think the code is flawed & would kick incorrectly on non-event reminders - but I didn't try to address that here as my focus is on trying to understand how `CRM_ACL_API::group` is called

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I need to re-check this - it looks like one check might be 'view' & then it further filters to 'edit' - closing for now

**Pull request closed without merging**